src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -0,0 +1,83 @@
+package com.codeus.winter.config;
+
+import com.codeus.winter.exception.BeansException;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultBeanFactory implements BeanFactory<Object> {
+    private final Map<String, Object> singletonBeans = new HashMap<>();
+    private final Map<String, BeanDefinition> beanDefinitions = new HashMap<>();
+    private final List<BeanPostProcessor> postProcessors = new ArrayList<>();
+
+
+    @Nullable
+    @Override
+    public Object getBean(String name) throws BeansException {
+        final Object bean = singletonBeans.get(name);
+        postProcessors
+                .forEach(postProcessor -> postProcessor.postProcessBeforeInitialization(bean, name));
+        postProcessors
+                .forEach(postProcessor -> postProcessor.postProcessAfterInitialization(bean, name));
+        return bean;
+    }
+
+    @Nullable
+    @Override
+    public Object getBean(String name, Class<Object> requiredType) throws BeansException {
+        return getBean(name);
+    }
+
+    @Nullable
+    @Override
+    public Object getBean(Class<Object> requiredType) throws BeansException {
+        return singletonBeans.entrySet().stream()
+                .map(Map.Entry::getValue)
+                .filter(requiredType::isInstance)
+                .findFirst()
+                .map(requiredType::cast)
+                .orElseThrow(() -> new BeansException("Bean not found for type: " + requiredType.getName()));
```

Comment:
we need to create for each specified case some specified exception
not found -> BeanNotFoundException
2 realization -> NotUniqueBeanDefinitionException

Reply:
I created an extra Exception - NotUniqueBeanDefinitionException and renamed BeansException to BeanNotFoundException.
After that, I fixed the logic in the methods according to the comment.